crates/cheatcodes/src/inspector.rs

```diff
@@ -1132,6 +1137,8 @@ impl Cheatcodes {
             }
         }
 
+        self.zk_should_update_nonce.take();
```

Comment:
We are already `take`ing in the `try_create_` method so this can be ignored. IMO it's cleaner to have zk related logic there, as much as possible to keep the upstream code unchanged.

Reply:
This is here to clear the flag when we are **not** running zk code, otherwise the flag would remain set to "update" also after the toplevel methods are called, since we avoid running them in EraVM